### PR TITLE
fix: jsonl type coercion

### LINKED ISSUE #59: bug: jsonl files returns all booleans value when `jsonl_type_coercion_strategy=any`
I accept the default `any` config for `jsonl_type_coercion_strategy` which creates a schema that has all available data types in it but the SDK [_conform_record_data_types](https://github.com/meltano/sdk/blob/fdd35a8f3584081e477d2bf0c20729b7cdf8ab52/singer_sdk/helpers/_typing.py#L390) ends up casting my records to the first listed type.

For example I sync this file it gets read properly but all of the values in the output are set to `true` because of the casting.

config:
```
{
    "protocol": "file",
    "file_path": ".secrets/data",
    "file_type": "jsonl"
  }
```

File content:
```
{"id": 1, "description": "Red-headed woodpecker", "verified": true, "views": 27, "created_at": "2021-09-22T01:01:05Z", "_sdc_extracted_at": "2023-08-30 19:41:39.464523+00:00", "_sdc_received_at": "2023-08-30 15:41:41.181791", "_sdc_batched_at": "2023-08-30 15:41:41.181777", "_sdc_deleted_at": null, "_sdc_sequence": 1693424501182, "_sdc_table_version": null}
...
``` 

Results:
```
{"type": "SCHEMA", "stream": "file", "schema": {"properties": {"id": {"type": ["null", "boolean", "integer", "number", "string", "array", "object"]}, "description": {"type": ["null", "boolean", "integer", "number", "string", "array", "object"]}, "verified": {"type": ["null", "boolean", "integer", "number", "string", "array", "object"]}, "views": {"type": ["null", "boolean", "integer", "number", "string", "array", "object"]}, "created_at": {"type": ["null", "boolean", "integer", "number", "string", "array", "object"]}, "_sdc_extracted_at": {"type": ["null", "boolean", "integer", "number", "string", "array", "object"]}, "_sdc_received_at": {"type": ["null", "boolean", "integer", "number", "string", "array", "object"]}, "_sdc_batched_at": {"type": ["null", "boolean", "integer", "number", "string", "array", "object"]}, "_sdc_deleted_at": {"type": ["null", "boolean", "integer", "number", "string", "array", "object"]}, "_sdc_sequence": {"type": ["null", "boolean", "integer", "number", "string", "array", "object"]}, "_sdc_table_version": {"type": ["null", "boolean", "integer", "number", "string", "array", "object"]}, "_sdc_file_name": {"type": "string"}, "_sdc_line_number": {"type": "integer"}, "_sdc_last_modified": {"format": "date-time", "type": "string"}}}, "key_properties": [], "bookmark_properties": ["_sdc_last_modified"]}
2023-08-31 13:55:33,662 | INFO     | tap-universal-file   | Starting sync of /Users/pnadolny/Documents/Git/GitHub/MeltanoLabs/tap-universal-file/.secrets/data/data.jsonl.
{"type": "RECORD", "stream": "file", "record": {"id": true, "description": true, "verified": true, "views": true, "created_at": true, "_sdc_extracted_at": true, "_sdc_received_at": true, "_sdc_batched_at": true, "_sdc_deleted_at": null, "_sdc_sequence": true, "_sdc_table_version": null, "_sdc_file_name": "/Users/pnadolny/Documents/Git/GitHub/MeltanoLabs/tap-universal-file/.secrets/data/data.jsonl", "_sdc_line_number": 1, "_sdc_last_modified": "2023-08-31T17:46:00+00:00"}, "time_extracted": "2023-08-31T17:56:09.764259+00:00"}
```

Using `jsonl_type_coercion_strategy=string` works as expected.

### PR DESCRIPTION
**Pull request by sebastianswms**
Closes #59 

I also fixed a bug with the `additional_info` configuration option. Setting it to false (to aid data validation in `test_jsonl_execution_coercion_any()`) was causing:

```
singer_sdk.helpers._typing.EmptySchemaTypeError: Could not detect type from empty type_dict. Did you forget to define a property in the stream schema?
```

### CODE REVIEW

File: tap_universal_file/client.py [hunk 1]
```diff
@@ -17,6 +18,41 @@
     from singer_sdk.tap_base import Tap
 
 
+def patched_is_boolean_type(property_schema: dict) -> bool | None:
```
Comment (visch) [inline tap_universal_file/client.py:21]: @edgarrmondragon  do we want to just get this into the SDK? 

We did something similar here in tap-postgres

https://github.com/MeltanoLabs/tap-postgres/blob/47b61b6c018c8272cec23f39244c5961bb730b72/tap_postgres/client.py#L34-L54

**Review by visch — APPROVED** [review]
@edgarrmondragon  can you take a peek at this

File: tap_universal_file/client.py (hunk 1 above)
Comment (edgarrmondragon) [reply tap_universal_file/client.py:21]: Yeah, I think we need to fix this upstream in the SDK.

**Review by edgarrmondragon — APPROVED** [review]

File: tap_universal_file/client.py (hunk 1 above)
Comment (sebastianswms) [reply tap_universal_file/client.py:21]: I'll create an issue for it.

Comment (sebastianswms) [reply tap_universal_file/client.py:21]: https://github.com/meltano/sdk/issues/2553

**Pull request merged**